File: Salon_Mgmt_Backend/src/main/java/com/salon/services/UserService.java
package com.salon.services;

import java.util.List;

import org.springframework.web.multipart.MultipartFile;

import com.salon.dtos.ApiResponse;
import com.salon.dtos.ChangePasswordDTO;
import com.salon.dtos.LoginRequestDTO;
import com.salon.dtos.LoginResponseDTO;
import com.salon.dtos.ProfileDTO;
import com.salon.dtos.UpdateProfileDTO;
import com.salon.dtos.UserRegisterDTO;
import com.salon.dtos.UserResp;

public interface UserService {
	List<UserResp> getAllUsers();
	
	ApiResponse registerCustomer(UserRegisterDTO user);
	ApiResponse registerOwner(UserRegisterDTO user);
	
	LoginResponseDTO login(LoginRequestDTO dto);
	ApiResponse deleteUser(Integer id);
	
	 // 🔹 UPDATE PROFILE (name, phone)
    ApiResponse updateProfile(Integer id, UpdateProfileDTO dto);

    // 🔹 CHANGE PASSWORD
    ApiResponse changePassword(Integer id, ChangePasswordDTO dto);

    // 🔹 UPLOAD PROFILE IMAGE
    ApiResponse uploadProfileImage(Integer id, MultipartFile file);
    
    ProfileDTO getProfile(Integer id);
}

File: frontend/src/services/userService.js
import axios from "./api";

/* ================= AUTH ================= */

// LOGIN
export const loginUser = (data) => {
  return axios.post("/users/login", data);
};

// CUSTOMER REGISTRATION
export const registerCustomer = (data) => {
  return axios.post("/users/register/customer", data);
};

// OWNER REGISTRATION
export const registerOwner = (data) => {
  return axios.post("/users/register/owner", data);
};

/* ================= USERS ================= */

// GET ALL USERS (Admin)
export const getAllUsers = () => {
  return axios.get("/users/getUsers");
};

// DELETE USER
export const deleteUser = (id) => {
  return axios.delete(`/users/removeUser/${id}`);
};

/* ================= PROFILE ================= */

export const getProfile = () =>
  axios.get(`/users/profile`);

export const updateProfile = (id, data) =>
  axios.put(`/users/profile/${id}`, data);

export const changePassword = (id, data) =>
  axios.put(`/users/change-password/${id}`, data);

export const uploadProfileImage = (id, file) => {
  const formData = new FormData();
  formData.append("image", file);
  return axios.post(`/users/upload-image/${id}`, formData);
};
